### ISSUE TITLE
Questions: Passing params to ko-component-routers & child routers

### ISSUE THREAD

**Issue by carlosagsmendes** [Profiscience/ko-component-router#61]
Hi @caseyWebb, I have two doubts in regards to the component router

1) I need to share data between my base component (loaded by knockout) and the components loaded by the router. 

In my _app_ I pass some _params_ to my main component: 

`<app params="myAppResources: textResource"></app>`

When working with parent and child routers I can access the parent's data that through the `ctx.$parent` on the child router.

But I'm not finding a way to access the params from the app component in the components loaded with  ko-component-router (i.e., how can I access to the _myAppResources_ from the Home or Tasks component?)     

Here's a [sample](https://jsfiddle.net/9sugpL9y/37/) running on JSFiddle

2) In the previous JSFiddle sample, the tasks component has a child router. I want to show the details for tasks. My routes and links seem to be fine but I'm not able to get the TaskDetails component to load when clicking on a task. What I'm I missing here? 

Thanks in advance!

**Comment by carlosagsmendes** [comment]
There was a typo on the JSFiddle. I've just updated the link above.

**Comment by carlosagsmendes** [comment]
In regards to 2), I've updated the sample JSFiddle again to include some logging when loading components. 

When I navigate to the tasks route the path, pathname and canonicalPath are not updated. 
When I navigate back to home the pathm pathname and canonicalPath are updated but they have the data relative to the previous route: `Component Name: home Path: /tasks PathName: /tasks CanonicalPath: /9sugpL9y/36/show/tasks`  

![myImage](http://content.screencast.com/users/carlos.mendes/folders/Jing/media/de58d925-bc29-4382-8067-5a3343dd1138/00000683.png)

**Comment by barsh** [comment]
@carlosagsmendes, generally we try to avoid attaching things to `window`, but it's ok to attach app globals to `window` like this:

`window.globals = { myAppResources: 'text' }`

or
`window.globals = { myAppResources: ko.observable('text') }`

that's what we do

**Comment by caseyWebb** [comment]
You've got some syntax errors in your fiddle. I've fixed them and turned it into little module (fiddle is a pita to work with re: routing). I've also added a bit to highlight how to pass data between components (although I found some other bugs in doing so).

https://gist.github.com/caseyWebb/da69af8674744c9d8b4d9a41cfc1684e

_use `npm install && npm start` to run it_

`ctx.state()` should be used to pass data between levels, but there is currently no way to pass in data to the top component. Generally this data would be fine living either on window or in a singleton module.

i.e.
_myAppResources.js_

``` javascript
'use strict'

module.exports = ko.observable()
```

_app.js_

``` javascript
'use strict'

const myAppResources = require('./myAppResources')
myAppResources('text')

// ...router stuff
```

_someRoutedPage.js_

``` javascript
'use strict'

const myAppResources = require('./myAppResources')
myAppResources()
// text
```

**Comment by carlosagsmendes** [comment]
Thank you for the suggestion @barsh. We use something similar to what Casey suggested (require through systemJS).  

@caseyWebb, thanks so much for your support.

From your changes, my understanding is that currently:

1) [I cannot use synchronous components](https://gist.github.com/caseyWebb/da69af8674744c9d8b4d9a41cfc1684e#file-app-js-L49) on child routers even though this was corrected and tested with your example on  #60 

2) To support the Tasks UI with the List on a component and the details on another I have to add an [empty dummy component to the '/' route](https://gist.github.com/caseyWebb/da69af8674744c9d8b4d9a41cfc1684e#file-app-js-L54) on the child router. I tried to remove the '/' route, but it didn't work.

3) In regards to passing state to child routers, we have to [do it on the View](https://gist.github.com/caseyWebb/da69af8674744c9d8b4d9a41cfc1684e#file-app-js-L72). There's no way to pass state in the viewModel, right?

Also, the state is passed as a computed property so it won't work in the Task->TaskDetails model because it is read only.  Since I need to edit the Task on the TaskDetails component, but I'm editing an observable from the task list on the Tasks component I need use ctx.$parent to access to my parent router and work with the original data. Is this ok?  

4) You mentioned that there was a bug and I should be able to use just ['/taskid' instead of '/tasks/taskId' on my paths](https://gist.github.com/caseyWebb/da69af8674744c9d8b4d9a41cfc1684e#file-app-js-L71), in my child router

```
 constructor(ctx) {
          // if `'/' + task.taskId` instead of `'/tasks/' + task.taskId` were used
          // then `$parent` would be unecessary. With that baing said, we should
          // probably pass state down to the bottom router.
            this.id = ko.pureComputed(() => ctx.$parent.state() && ctx.$parent.state().taskId)
        }
```

When this is corrected will I still be able to access the ctx.$parent router? I'm asking because as explained since the state is read-only it won't work for the Master->Details model that I'm using. 

Thanks again
